pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/ComputeFieldStep.java

```diff
@@ -148,12 +154,66 @@ private GenericRecord computeFields(
     }
     // Add computed fields
     for (ComputeField field : fields) {
-      newRecordBuilder.set(field.getName(), field.getEvaluator().evaluate(context));
-      ;
+      newRecordBuilder.set(
+          field.getName(),
+          getAvroValue(
+              newSchema.getField(field.getName()).schema(),
+              field.getEvaluator().evaluate(context)));
     }
     return newRecordBuilder.build();
   }
 
+  private Object getAvroValue(Schema schema, Object value) {
+    if (value == null) {
+      return null;
+    }
+
+    LogicalType logicalType = getLogicalType(schema);
+    if (logicalType == null) {
+      return value;
+    }
+
+    // Avro logical type conversion: https://avro.apache.org/docs/1.8.2/spec.html#Logical+Types
+    switch (logicalType.getName()) {
+      case "date":
+        validateLogicalType(value, schema.getLogicalType(), LocalDate.class);
+        LocalDate localDate = (LocalDate) value;
+        return (int) localDate.toEpochDay();
+      case "time-millis":
+        validateLogicalType(value, schema.getLogicalType(), LocalTime.class);
+        LocalTime localTime = (LocalTime) value;
+        return (int) (localTime.toNanoOfDay() / 1000000);
+      case "timestamp-millis":
+        validateLogicalType(value, schema.getLogicalType(), LocalDateTime.class);
```

Comment:
Shouldn't it be mapped to a j.u.Instant ?

Reply:
This is driven by the type generated from the JSTL evaluator. Now the same question applies (why not use Instant with ejstl?)

We could, I just found it:
1. More consistent with the Time and Date types (because for those the better option is LocalTime & LocalDate as those parse the provided input in ISO format parts out of the box - thinks like `2017-02-10` or `22:30:00`
2. Instant iso format require the user to pass empty Z in the format (like `2017-02-10T22:30:00Z`, but local date time accepts `2017-02-10T22:30:00`. As far as user is concerned, all dates in their avro output will be in reference to epoch
3. LocalDateTime also provides `plus` operation with Units of years and months which as used with the `dateadd` method. Instant only accepts those values: https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#plus-long-java.time.temporal.TemporalUnit-

Reply:
It seems to me that if the datetime is referenced to epoch, Instant is the type to use whereas LocalDatetime is not bound to a timezone (it's the time you see on the clock on the wall without knowing where you are). But probably you transparently assume UTC timezone and the user doesn't see the LocalDatetime type anyway.

Reply:
that is fair, let me switch to time instant. And then we can decide separately if we want the user format to be:
`'2007-12-03T10:15:30` vs `2007-12-03T10:15:30.00Z` (the latter is the Instant requirement, empty Z must be included indicating UTC.

Either way, the user is not aware of the underlying date object - it is just easier for us to use the underlying parse method as is without pre-processing

Reply:
I don't really know why Instant forces the Z part since it is always strictly UTC!

Reply:
OK I see. What I would suggest is that we support [RFC3339](https://www.rfc-editor.org/rfc/rfc3339#section-5.6) dates. This maps well to j.u.t.OffsetDateTime.
With maybe some tolerance on the input string where if the time offset is not there, we assume UTC. I think Jackson does it (not sure).

Reply:
Also OffsetDateTime has `plusMonths` and `plusYears`

Reply:
+1 to ISO 3330. to make documentation easier, we can stick to its format and refer user to the RFC. (for UTC, they could write either 2022-07-02T01:02:00Z OR 2022-07-02T01:02:00+00). If the provide millis (which I think will be common case) we assume epoch millis. Another common use case that won't include the string is fn:now()).

we can always add more permissive format (like 2022-07-02T01:02:00) without breaking existing users...